lib/less/tree/ruleset.js

```diff
@@ -182,15 +182,15 @@ Ruleset.prototype.evalImports = function(context) {
     }
 };
 Ruleset.prototype.makeImportant = function() {
-    return new Ruleset(this.selectors, this.rules.map(function (r) {
-                if (r.makeImportant) {
-                    return r.makeImportant();
-                } else {
-                    return r;
-                }
-            }), this.strictImports);
-};
-Ruleset.prototype.matchArgs = function (args) {
+    this.rules = this.rules.map(function (r) {
```

Comment:
This would fail:
````
size(@aaa) {
  .set-width(@aaa) {
    width: @aaa; // this line is failing
  }
  .set-width(@aaa);
}
.when-calling-nested-issue-2394 {
  .size(10px) !important;
}
````

with:
````
NameError: variable @aaa is undefined in c:\data\meri\less.js\tests\test.less on line 3, column 12:
2   .set-width(@aaa) {
3     width: @aaa;
4   }
````

I could not figure why. I tried to find missing property in new ruleset, compare new properties with old ones etc. Since the method was called only from one place, I  gave up and did what worked.

This test case is inside mixins-important.less file.